packages/dgraph/src/client.ts

```diff
@@ -0,0 +1,58 @@
+import * as jwt from "jsonwebtoken"
+import fetch, { HeadersInit } from "node-fetch"
+
+export interface DgraphClientParams {
+  endpoint: string
```

Comment:
I believe for Dgraph cloud there needs to be an option for an apiKey which can be set https://dgraph.io/docs/cloud/admin/authentication/. What you have here looks great for self managed and local deploys of DGraph but without the client apiKey option which would be set under the header `Dg-Auth: <your-api-key>' ` this will not work for those folks using DGraph cloud as the queries made by non authenticated users (ie: login, register etc.) won't be available to anonymous users of a secured schema.

Reply:
I don't believe the current implementation is secure https://github.com/nextauthjs/adapters/blob/next/packages/dgraph/src/graphql/secure.schema.gql#L44. Can someone explain how `$nextAuth` protects the queries/mutations, and who sets this value?

Reply:
For reference I typically treat the "system" or "admin" as I refer to it as with a separate secret that gets injected into the schema which looks like this:

```graphql
type Account
  @auth(
    delete: { rule: "{$adminSecret: { eq: \"<ADMIN_SECRET>\" } }" }
    add: { rule: "{$adminSecret: { eq: \"<ADMIN_SECRET>\" } }" }
    query: { rule: "{$adminSecret: { eq: \"<ADMIN_SECRET>\" } }" }
    update: { rule: "{$adminSecret: { eq: \"<ADMIN_SECRET>\" } }" }
  ) {}
``
and declare the env in `pages/api/[...nextauth]`
```tsx
export default NextAuth({
  adapter: DgraphAdapter({
    endpoint: process.env.NEXT_PUBLIC_GRAPHQL_URL!,
    apiKey: process.env.GRAPHQL_API_KEY,
    adminSecret: process.env.ADMIN_SECRET,
    authHeader: process.env.AUTH_HEADER,
    jwtSecret: process.env.JWT_SECRET,
  }),
})
```

That way there is a secure comparison between a client request and the graphql api in Dgraph. As far as I can see, `$nextAuth` will always resolve true and allow any request access to all values in the secure schema, rendering it the same as the unsecure variant.

Reply:
Here: https://github.com/nextauthjs/adapters/blob/4c9b4376189e4aa60869537b10208506ab166d4f/packages/dgraph/src/client.ts#L35-L37

We generate the content of the JWT that is sent to Dgraph. In our case, the content is `nextAuth: true`, which signals Dgraph that we have the correct "role" to execute operations like this:

https://github.com/nextauthjs/adapters/blob/4c9b4376189e4aa60869537b10208506ab166d4f/packages/dgraph/src/graphql/secure.schema.gql#L3

Reply:
Just read through this prior to you answering, retracted those statements as this is correct. Awesome :+1:

Reply:
I think as long as we can verify this configuration is valid for DGraph cloud, this is great.

Reply:
docs updated nextauthjs/docs#120